xcode/Subconscious/Shared/Services/UserProfileService.swift

```diff
@@ -103,13 +183,25 @@ actor UserProfileService {
             throw UserProfileServiceError.invalidSphereIdentity
         }
         
+        let address = Slashlink.ourProfile.toPublicMemoAddress()
+        let userProfileData = await readProfile(address: address)
+        
+        let pfp: ProfilePicVariant = Func.run {
+            if let url = URL(string: userProfileData?.profilePictureUrl ?? "") {
+                return .url(url)
+            }
+            
+            return .none
+        }
+        
+        
         let profile = UserProfile(
             did: did,
-            petname: petname,
-            address: Slashlink.ourProfile.toPublicMemoAddress(),
-            // TODO: replace with _profile_.json data
-            pfp: "sub_logo",
-            bio: "Wow, it's your own profile! With its own codepath!",
+            petname: Petname(userProfileData?.nickname ?? "") ?? nickname,
```

Comment:
Something to think about: the nickname stored in `AppDefaults` and the nickname in `_profile_` can diverge. Should we even be storing nickname in `AppDefaults` going forward?

Reply:
Probably not. Sphere should be sufficient.

We also store an `ownerKeyName` in AppDefaults. This happens to be the same as the `nickname` right now.

@cdata would it be a good idea to specify a hard-coded owner key name for spheres, such as `user`, instead?

Reply:
Let's deal with this in a followup.